src/js/components/TextInput/index.d.ts

```diff
@@ -12,8 +12,8 @@ export interface TextInputProps {
   messages?: {enterSelect?: string,suggestionsCount?: string,suggestionsExist?: string,suggestionIsOpen?: string};
   name?: string;
   onSelect?: ((x: { target: React.RefObject<HTMLElement>['current'], suggestion: any }) => void);
-  onSuggestionsOpen?: ((...args: any[]) => any);
-  onSuggestionsClose?: ((...args: any[]) => any);
+  onSuggestionsOpen?: (() => Function);
+  onSuggestionsClose?: (() => Function);
```

Comment:
This suggested type wouldn't work on the code example of https://storybook.grommet.io/?path=/story/textinput--custom

Reply:
@ShimiSun why wouldnt it work? I tested this here and it worked the same way

Reply:
Sorry, I didn't understand. I tested it manually and it seemed to work the same way it does in the link, or in the master branch :thinking: 
Could you explain, please? :)

Reply:
Are you sure? this is what I'm getting when adding Custom.tsx file with the content of https://storybook.grommet.io/?path=/story/textinput--custom and using the suggested fix
![image](https://user-images.githubusercontent.com/6320236/76438170-43a3e680-6380-11ea-8497-526742550201.png)
```
Type 'void' is not assignable to type 'Function'.ts(2322)
```

Reply:
Oh. Now I see it. Just made a commit with the changes, everything seems fine now :)
Thanks again

Reply:
Can you please add a tsx story file that can validate this change? similarly to what I've shown above.

Reply:
Sure. I submmited the commit before I saw your message on the other PR. I'll add in both

Reply:
The file is now added :)